packages/devtools_app/lib/src/screens/deep_link_validation/project_root_selection/select_project_view.dart

```diff
@@ -59,25 +60,47 @@ class _SelectProjectViewState extends State<SelectProjectView>
     });
   }
 
-  void _handleValidateProject(String directory) async {
-    setState(() {
-      _retrievingFlutterProject = true;
-    });
+  Future<List<String>> _requestAndridVariants(String directory) async {
     ga.timeStart(gac.deeplink, gac.AnalyzeFlutterProject.loadVariants.name);
     final List<String> androidVariants =
         await server.requestAndroidBuildVariants(directory);
-    if (!mounted) {
+    if (androidVariants.isEmpty) {
+      // If the project is not a Flutter project, cancel timing and return an empty list.
       ga.cancelTimingOperation(
         gac.deeplink,
         gac.AnalyzeFlutterProject.loadVariants.name,
       );
+      return [];
+    } else {
+      ga.timeEnd(gac.deeplink, gac.AnalyzeFlutterProject.loadVariants.name);
+      return androidVariants;
+    }
+  }
+
+  Future<XcodeBuildOptions> _requestiOSBuildOptions(String directory) async {
+    ga.timeStart(
+      gac.deeplink,
+      gac.AnalyzeFlutterProject.loadIosBuildOptions.name,
+    );
+    final XcodeBuildOptions iosBuildOptions =
+        await server.requestIosBuildOptions(directory);
+    ga.timeEnd(
```

Comment:
do we need the same logic to check if the iosBuildOptions are empty like we have to the android variants?

Reply:
i think it's fine, the check here is just to check if the directory is a flutter project, check them in either Android/iOS for once should be enough, no need to do it twice.